### DISCUSSION TITLE
RuntimeError: wrap_controller at 'pytest_runtest_makereport' dash/testing/plugin.py:106 did not yield

### DISCUSSION THREAD

**Discussion by alexander-soare** [pytest-dev/pytest#10820] [Q&A] [answered]
Our CI suddenly started encountering this error (during Pytest's collection phase) for all PRs and I can't reproduce it locally. I'm running these on Ubuntu20.04 in a Docker container (also Ubuntu20.04). To be clear, CI has been running fine, and it's no one particular PR that's doing this today, it's all of them, regardless of the content. Any ideas?

I tried googling and couldn't find a single instance of this error.

I'm using Python3.10 and installing pytest==7.2.0. I can provide a stack trace upon request.

**Comment by alexander-soare** [comment]
Potential clue: I have a pip cache on my Docker base image so I know which dependencies are new by looking at the pip install output. I can tell a newer version of dash is being downloaded:

```
Collecting dash>=2.6.0
  Downloading dash-2.9.0-py3-none-any.whl
```

**Comment by RonnyPfannschmidt** [comment, marked as answer]
https://github.com/plotly/dash/blob/dev/dash/testing/plugin.py#L108 indicates a bug in their code

**Reply by alexander-soare** [reply]
Is Dash a direct dependency of Pytest (actually I couldn't find it but perhaps I didn't check the right files). If it were, would it be best to patch Pytest 2.7 such that it makes sure the Dash version is <=2.8?

**Reply by RonnyPfannschmidt** [reply]
no - https://github.com/plotly/dash/pull/2417 broke stuff

**Reply by RonnyPfannschmidt** [reply]
https://github.com/plotly/dash/pull/2461/files will fix

**Reply by fraghag** [reply]
Fixed in Dash 2.9.1
